packages/compass-sidebar/src/components/multiple-connections/sidebar.tsx

```diff
@@ -91,12 +151,23 @@ export function MultipleConnectionSidebar({}: MultipleConnectionSidebarProps) {
           favoriteConnections={favoriteConnections}
           nonFavoriteConnections={recentConnections}
           onConnect={onConnect}
-          onNewConnection={onNewConnection}
+          onNewConnection={() => setIsConnectionFormOpen(true)}
```

Comment:
We use `useCallback()` for other callbacks in this component, should we do it for this one as well? Otherwise it's kind of removing the benefits of the other `useCallback()`s, right?